packages/legend-query-builder/src/components/filter/QueryBuilderFilterPanel.tsx

```diff
@@ -247,9 +273,9 @@ export const buildFilterTreeWithExists = (
       const parentPropertyChainIndex = existsLambdaPropertyChains.findIndex(
         (p) =>
           p instanceof AbstractPropertyExpression &&
-          p.func.value ===
-            targetDropNode.propertyExpressionState.propertyExpression.func
-              .value &&
+          p.func.value.hashCode ===
+            targetDropNode.propertyExpressionState.propertyExpression.func.value
+              .hashCode &&
```

Comment:
I don't think we should use hashcode to check the equality. Do you know where the inequality is happening. I remember that it used to work without deepclone

Reply:
Here we are comparing two Property. they have same values but different reference.

Reply:
hashcode is generated upon  👀

Reply:
Got it. I just feel hashcode is something we use to do change detection. We generally don't use it do these kinds of equality checks across the codebase If deepClone is failing this check to preserve the reference of property we can try using `buildPropertyExpressionChain` function to build the new expression. This was you hold no references to the property expression present in project. @MauricioUyaguari  what do you think?